docs/Dockerfile

```diff
@@ -0,0 +1,132 @@
+# Copyright 2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+# Description:
+#   Builds the environment needed to build the Sawtooth Seth docs
+#   Running the image will put the Sawtooth Seth docs in
+#   sawtooth-seth/docs/build on your local machine.
+#
+# Build:
+#   $ cd sawtooth-seth
+#   $ docker build . -f docs/Dockerfile -t seth-build-docs
+#
+# Run:
+#   $ cd sawtooth-seth
+#   $ docker run -v $(pwd)/docs:/project/sawtooth-seth/docs seth-build-docs
+
+FROM ubuntu:xenial
+
+COPY . /project/sawtooth-seth
+
+ENV GOPATH=/project/sawtooth-seth/common
+ENV PATH=$PATH:/project/sawtooth-seth/common/bin:/project/sawtooth-seth/bin:/usr/lib/go-1.9/bin
+
+# Build cli and tp
+RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 8AA7AF1F1091A5FD \
+ && echo 'deb http://repo.sawtooth.me/ubuntu/1.0/stable xenial universe' >> /etc/apt/sources.list \
+ && apt-get update \
+ && apt-get install -y -q \
+    git \
+    golang-1.9-go \
+    libssl-dev \
+    libzmq3-dev \
+    openssl \
+    python3 \
+    python3-grpcio-tools=1.1.3-1 \
+    python3-pip \
+    python3-sawtooth-cli \
+    software-properties-common \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+RUN go get -u \
+    github.com/golang/protobuf/proto \
+    github.com/golang/protobuf/protoc-gen-go \
+    github.com/btcsuite/btcd/btcec \
+    github.com/rberg2/sawtooth-go-sdk/... \
+    github.com/jessevdk/go-flags \
+    github.com/pelletier/go-toml \
+    golang.org/x/crypto/ssh/terminal \
+    golang.org/x/crypto/ripemd160 \
+    gopkg.in/fatih/set.v0
+
+RUN seth-protogen go
+
+ENV PATH=$PATH:/project/sawtooth-seth/cli/bin
+ENV GOPATH=$GOPATH:/project/sawtooth-seth/burrow:/project/sawtooth-seth/cli
+WORKDIR /project/sawtooth-seth/cli/src/seth_cli/cli
+RUN go build -o /project/sawtooth-seth/cli/bin/seth
+
+ENV PATH=$PATH:/project/sawtooth-seth/processor/bin
+ENV GOPATH=$GOPATH:/project/sawtooth-seth/processor
+WORKDIR /project/sawtooth-seth/processor/src/seth_tp
+RUN go build -o /project/sawtooth-seth/processor/bin/seth-tp
+
+# Build RPC
```

Comment:
I don't like the fact we have to repeat ourselves here, but until I have finished creating multi-stage Dockerfiles for the other components we can live with this for now...

Reply:
Agreed